feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/widget/SessionsItemDecoration.kt

```diff
@@ -54,35 +56,102 @@ class SessionsItemDecoration(
                 }
             }
 
+            val startDateTimeText =
+                calcDateTimeText(position, view, shouldShowDateText(position))
+
+            if (startDateTimeText.dateText != null) {
+                c.drawText(
+                    startDateTimeText.dateText.value,
+                    startDateTimeText.dateText.positionX,
+                    startDateTimeText.dateText.positionY,
+                    textPaint
+                )
+            }
+
             c.drawText(
-                startTimeText.value,
-                startTimeText.positionX,
-                startTimeText.positionY,
+                startDateTimeText.startTimeText.value,
+                startDateTimeText.startTimeText.positionX,
+                startDateTimeText.startTimeText.positionY,
                 textPaint
             )
         }
     }
 
-    private fun calcTimeText(position: Int, view: View): StartTimeText {
+    private fun shouldShowDateText(position: Int): Boolean {
+        if (!visibleSessionDate) {
+            return false
+        }
+        if (position <= 0) {
+            return true
+        }
+        val currentSessionItem = adapter.getItem(position) as SessionItem
+        val lastSessionItem = adapter.getItem(position - 1) as SessionItem
+        val isSameDateWithLastItem =
+            currentSessionItem.startSessionDate() == lastSessionItem.startSessionDate()
+        return when {
+            isSameDateWithLastItem -> false
+            else -> true
```

Comment:
When multiple same time's favorite session has, date time text will dismiss on scrolling one item height in this code.😭
But, I was not be able to fix this problem. I'll issue for the problem once.
![fps_30](https://user-images.githubusercontent.com/27717556/74103524-f4f00a80-4b8f-11ea-8fe2-3af11aa303b8.gif)